src/V3Global.cpp

```diff
@@ -34,6 +35,10 @@ AstNetlist* V3Global::makeNetlist() {
     return newp;
 }
 
+void V3Global::shutdown() {
+    VL_DO_DANGLING(delete m_hierPlanp, m_hierPlanp);  // delete nullptr is safe
+}
```

Comment:
```suggestion
void V3Global::shutdown() {
    VL_DO_CLEAR(delete m_hierPlanp, m_hierPlanp = NULL);  // delete nullptr is safe
}
```

Just a bit safer in case shutdown is mis-called.

Probably would be better to have this in destructor (then DO_DANGING would be correct).

Reply:
Thanks. I did not realized that VL_DANGLING is empty in C++03 setting.

`V3Global` is used only as a global variable `v3Global`.
I personally don't want to release resources in dtor of global/static variables because
- It is hard to control the order of destruction of multiple global/static objects (kind of racing)
- If trouble happens in a dtor of global/static object, debugging is harder than called from normal context.

So I like to keep this shutdown function. On the other hand, v3Global has been safely destructed in program epilogue.

If you still prefer deleting m_hierPlanp in dtor, I will do so.

Reply:
That's fine.  BTW VL_DANGLING is the way it is in order to make both formal tools happy and also avoid warnings about setting a variable that's never later used.  I wish C++ allowed "m_old_var = delete" to make it illegal to later use something.